src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -101,27 +119,51 @@ public void setMax(int max) {
     }
 
     @Override
+    public void beginTree(DetailAST rootAST) {
+        rangeValues.clear();
+        expressionValues.clear();
+        processedExpressionRange.resetRange();
+        currentRangeValue = INITIAL_VALUE;
```

Comment:
please cleanup `belongsToRangeValues`.
All class fields (except user defined properties) should be cleaned, to clear state completely for next file processing.